File: android/rctmapboxar/src/main/java/com/mapbox/react/ar/MapboxManager.java
package com.mapbox.react.ar;

public class MapboxManager {
    private static MapboxManager INSTANCE;
    private String accessToken;

    private MapboxManager(String accessToken) {
        this.accessToken = accessToken;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public static MapboxManager getInstance() {
        return getInstance(null);
    }

    public static MapboxManager getInstance(String accessToken) {
        if (INSTANCE == null && accessToken != null && !accessToken.isEmpty()) {
            INSTANCE = new MapboxManager(accessToken);
        }
        return INSTANCE;
    }
}

File: ios/RNMapboxAR/MapboxManager.m
//
//  MapboxManager.m
//  RNLocateTest
//
//  Created by Dave Prukop on 4/30/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "MapboxManager.h"

@implementation MapboxManager

static MapboxManager *INSTANCE = nil;

@synthesize accessToken;

+ (MapboxManager *)getInstance
{
  if (INSTANCE == nil) {
    INSTANCE = [[MapboxManager alloc] init];
  }
  return INSTANCE;
}

@end
